litestar/stores/redis.py

```diff
@@ -30,6 +33,7 @@ def __init__(self, redis: Redis, namespace: str | None | EmptyType = Empty) -> N
                 ``None``. This will make :meth:`.delete_all` unavailable.
         """
         self._redis = redis
+        self._should_close_store_manually = True
```

Comment:
Maybe have this as a parameter in `__init__`, so you don't need to patch this from the outside?